test/snippet/core/metafunction/deferred_crtp_base.cpp

```diff
@@ -0,0 +1,49 @@
+//! [code]
+#include <string>
+#include <type_traits>
+
+#include <seqan3/core/metafunction/deferred_crtp_base.hpp>
+
+// Defines a crtp_base class with an additional value type.
+template <typename derived_t, typename value_t>
+class base1
+{
+public:
+
+    value_t func1() const
+    {
+        return {"instance of base1"};
+    }
+};
+
+// Defines a crtp_base class with an additional return type and a parameter_t.
+template <typename derived_t, typename value_t, typename parameter_t>
+class base2
+{
+public:
+
+    value_t func2(parameter_t const p) const
+    {
+        return static_cast<value_t>(p);
+    }
+};
+
+// The derived class that inherits from a variadic crtp pattern.
```

Comment:
add
`// This declaration would not be possible without deferred crtp_base class instantiation as it would cause an incomplete type compiler error` (right?)
if I understood correct also add:
`// Note the derived class can now only be instantiated with deferred crtp base classes (created from normal crtp base classes by using seqan3::detail::deferred_crtp_base)`